src/peer/peer-listener.js

```diff
@@ -0,0 +1,44 @@
+const logger = require('../logger');
+
+const PEER_LISTENER_INTERVAL = process.env.PEER_LISTENER_INTERVAL || 5000; // 5 seconds
+
+class PeerListener {
+  constructor(peer, apiProxy) {
+    this.peer = peer;
+    this.apiProxy = apiProxy;
+  }
+
+  start() {
+    const self = this;
+    setInterval(async () => {
+      try {
+        const rooms = await self.apiProxy.listRooms();
```

Comment:
I'm not familiar with how the full flow works so maybe it's not relevant, but I think there is a possible race condition here if this call takes more than 5 secs, as the next tick of the interval callback would be triggered and might have conflicting actions.

Reply:
yep, I know what you're referencing here. That's not relevant here :)